ztp/gitops-subscriptions/argocd/SNOExpansion.md

```diff
@@ -0,0 +1,483 @@
+```bash
+$ cat <<EOF | oc apply -f -
+apiVersion: action.open-cluster-management.io/v1beta1
+kind: ManagedClusterAction
+metadata:
+  name: clear-ptp-node-selector
+  namespace: $SNO_NAME
+spec:                 
+  actionType: Delete
+  kube:             
+    resource: ptpoperatorconfigs
+    name: default
+    namespace: openshift-ptp
+EOF              
+```
+Monitor the outcome (within the next minute) by:
+
+```bash
+$ oc get managedclusteraction/clear-ptp-node-selector -n $SNO_NAME -o jsonpath='{.status.conditions[*]}'
+```
+Delete the managedclusterview to free up resources:
+
+```bash
+$ oc delete managedclusterview check-ptp-node-selector -n $SNO_NAME
+```
+##### __Ensuring SR-IOV daemon selector compatibility__
+Check SR-IOV operator `configDaemonNodeSelector`setting:
+```bash
+$ SNO_NAME=<your SNO cluster name here>
+
+$ cat << EOF | oc apply -f -
+apiVersion: view.open-cluster-management.io/v1beta1
+kind: ManagedClusterView
+metadata:
+  name: check-sriov-node-selector
+  namespace: $SNO_NAME
+spec:
+  scope:
+    resource: sriovoperatorconfigs
+    name: default
+    namespace: openshift-sriov-network-operator
+EOF
+```
+Check the operator config:
+
+```bash
+$ oc get managedclusterview check-sriov-node-selector -n $SNO_NAME -ojsonpath='{ .status.result.spec }'
+{"disableDrain":true,"enableInjector":true,"enableOperatorWebhook":true,"logLevel":2}
+```
+If there is `configDaemonNodeSelector`, and it is set to `node-role.kubernetes.io/master: ""`, it must be cleared.
+
+To clear the SR-IOV operator `configDaemonNodeSelector`, deploy the following manifest on your hub cluster:
+```bash
+$ cat <<EOF | oc apply -f -
+apiVersion: action.open-cluster-management.io/v1beta1
+kind: ManagedClusterAction
+metadata:
+  name: clear-sriov-node-selector
+  namespace: $SNO_NAME
+spec:                 
+  actionType: Delete
+  kube:             
+    resource: sriovoperatorconfigs
+    name: default
+    namespace: openshift-sriov-network-operator
+EOF              
+```
+
+Monitor the outcome (within the next minute) by:
+```bash
+oc get managedclusteraction/clear-sriov-node-selector -n $SNO_NAME -o jsonpath='{.status.conditions[*]}'
+{"lastTransitionTime":"2022-08-21T08:39:04Z","message":"Resource action is done.","reason":"ActionDone","status":"True","type":"Completed"}
+```
+
+Delete the managedclusterview to free up resources:
+
+```bash
+$ oc delete managedclusterview check-sriov-node-selector -n $SNO_NAME
+```
+##### __Redeploy SNO policies__
+Redeploy the common policies using ztp plugin container version 4.12 or above using TALM operator custom resource.
+
+#### __Preparing worker node policies__
+Create the following policy template:
+
+```yaml
+apiVersion: ran.openshift.io/v1
+kind: PolicyGenTemplate
+metadata:
+  name: "cnfdf15-worker"
+  namespace: "ztp-cnfdf15-policies"
+spec:
+  bindingRules:
+    # These policies will correspond to all clusters with this label:
+    sites: "cnfdf15"
+######
+# Comment 1
```

Comment:
Took me additional scrolling to realize these comments were below. Any reason to not include them inline?

Reply:
I was trying to do something similar to openshift documentation. I'll put them inline, I realize now that it is misleading